services/src/main/scala/cromwell/services/metadata/impl/builder/MetadataBuilderActor.scala

```diff
@@ -35,17 +35,38 @@ object MetadataBuilderActor {
   case object IdleData extends MetadataBuilderActorData
   final case class HasWorkData(target: ActorRef, originalRequest: BuildMetadataJsonAction)
       extends MetadataBuilderActorData
+
+  sealed trait EventsCollectorData extends MetadataBuilderActorData {
+    val target: ActorRef
+    val originalRequest: BuildMetadataJsonAction
+    val originalQuery: MetadataQuery
+    val originalEvents: Seq[MetadataEvent]
+    val subWorkflowsMetadata: Map[String, JsValue]
+    val waitFor: Int
+
+    def isComplete = subWorkflowsMetadata.size == waitFor
+  }
   final case class HasReceivedEventsData(target: ActorRef,
                                          originalRequest: BuildMetadataJsonAction,
                                          originalQuery: MetadataQuery,
                                          originalEvents: Seq[MetadataEvent],
                                          subWorkflowsMetadata: Map[String, JsValue],
                                          waitFor: Int
-  ) extends MetadataBuilderActorData {
+  ) extends EventsCollectorData {
     def withSubWorkflow(id: String, metadata: JsValue) =
       this.copy(subWorkflowsMetadata = subWorkflowsMetadata + ((id, metadata)))
+  }
 
-    def isComplete = subWorkflowsMetadata.size == waitFor
+  final case class HasReceivedCostEventsData(target: ActorRef,
```

Comment:
Previously we used `HasReceivedCostEventsData` as the data object when collecting per-subworkflow results for either plain metadata or cost. I'm splitting it into two classes here for the two different use cases, so we can include workflow status in the cost one. 

Opinions welcome on the naming - I find `HasReceivedCostEventsData` to be an unclear name for that object, but didn't want to introduce churn by changing it, so I tried to find a sensible-seeming name for a new parent trait.

Reply:
Honestly, i find that leaving an inline comment before the parent trait and then before the class to be more useful than trying to name things perfectly